Rename `[Simple|Aggregated]MetricSample` to `[Simple|Aggregated]MetricValue`

tiyash-basu-frequenz opened the pull request:
Closes #141

tiyash-basu-frequenz committed:
Rename `SimpleMetricSample` to `SimpleMetricValue`

In out terminology, we refer to a "sample" as a data structure that stores
a value alongside a timestamp. Since the message `SimpleMetricSample` does
not contain timestamps, it needs to be renamed.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Rename `AggregatedMetricSample` to `AggregatedMetricValue`

In out terminology, we refer to a "sample" as a data structure that stores
a value alongside a timestamp. Since the message `AggregatedMetricSample`
does not contain timestamps, it needs to be renamed.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

nhatcher-frequenz commented on proto/frequenz/api/common/v1/metrics/metric_sample.proto:
```diff
@@ -46,19 +46,19 @@ message AggregatedMetricSample {
+// A `MetricSampleVariant` can hold either a `SimpleMetricValue` or an
+// `AggregatedMetricValue`, but not both simultaneously. Setting one will
 // nullify the other.
 message MetricSampleVariant {
```
This needs to be renamed to `MetricValueVariant` and `metric_sample_type` to `metric_value_type`

tiyash-basu-frequenz replied:
Good catch! :bow: 

Fixed

nhatcher-frequenz reviewed (changes requested):
Minor issue

tiyash-basu-frequenz committed:
Rename `MetricSampleVariant` to `MetricValueVariant`

This variants present in this enum are not called samples anymore, but
values.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Rename `MetricSample.sample` to `MetricSample.value`

The datatype of this field is now `MetricValueVariant`, so this commit
tries to rename the field to a more appropriate one.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

nhatcher-frequenz commented on RELEASE_NOTES.md:
```diff
@@ -30,6 +30,16 @@
+
+- Renamed `MetricSampleVariant` to `MetricValueVariant`.
+
+- Rename `MetricSample.sample` to ``MetricSample.value`.
```
Just one back tick.

tiyash-basu-frequenz committed:
Rename `SensorMetricSample.sample` to `SensorMetricSample.value`

The datatype of this field is now `MetricValueVariant`, so this commit
tries to rename the field to a more appropriate one.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

nhatcher-frequenz reviewed (approved):
LGTM!
